catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/ProcessResourceImpl.java

```diff
@@ -168,8 +179,16 @@ public String getMimeType() {
   }
 
   @Override
-  public InputStream getInputStream() throws IOException {
-    return inputStream;
+  public synchronized InputStream getInputStream() throws IOException {
+    if (inputStreamDataCache == null) {
+      inputStreamDataCache = new TemporaryFileBackedOutputStream(32 * 1024);
+      IOUtils.copyLarge(inputStream, inputStreamDataCache);
```

Comment:
❗️ `inputStream` should be closed here once we are done copying it as opposed to waiting at the end when the resource is closed.